frontend/endpoints/destinations.go

```diff
@@ -56,6 +58,21 @@ type Destination struct {
 	Conditions      []metav1.Condition           `json:"conditions,omitempty"`
 }
 
+var _ config.ExporterConfigurer = &Destination{}
+
+func (dest Destination) GetID() string {
+	return dest.Name
+}
+func (dest Destination) GetType() common.DestinationType {
+	return dest.Type
+}
+func (dest Destination) GetConfig() map[string]string {
+	return dest.Fields
+}
+func (dest Destination) GetSignals() []common.ObservabilitySignal {
+	return exportedSignalsObjectToSlice(dest.ExportedSignals)
+}
```

Comment:
This already exists [here](https://github.com/odigos-io/odigos/blob/b2f196915907bf5f7460fd02d22df9e3ab6bc2bd/api/odigos/v1alpha1/destination_types.go#L67).

Is defining it again intentional?

Reply:
Yes, these are for the ui struct and those you linked are for the k8s one - they both implement this interface, and it is used when we call `ModifyCOnfig` at the `TestConnection` function

Reply:
Right!
so confusing 😅

LGTM